presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveDynamicPartitionPruning.java

```diff
@@ -0,0 +1,219 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.Session;
+import io.prestosql.operator.OperatorStats;
+import io.prestosql.spi.QueryId;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.ValueSet;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.MaterializedResult;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.testing.ResultWithQueryId;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.airlift.testing.Assertions.assertGreaterThanOrEqual;
+import static io.airlift.testing.Assertions.assertLessThan;
+import static io.airlift.units.Duration.nanosSince;
+import static io.prestosql.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.prestosql.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.prestosql.server.DynamicFilterService.DynamicFilterDomainStats;
+import static io.prestosql.server.DynamicFilterService.DynamicFiltersStats;
+import static io.prestosql.spi.predicate.Domain.none;
+import static io.prestosql.spi.predicate.Domain.singleValue;
+import static io.prestosql.spi.predicate.Range.range;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static io.prestosql.tpch.TpchTable.getTables;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestHiveDynamicPartitionPruning
+        extends AbstractTestQueryFramework
+{
+    private static final Logger log = Logger.get(TestHiveDynamicPartitionPruning.class);
+    private static final String PARTITIONED_LINEITEM = "partitioned_lineitem";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setInitialTables(getTables())
+                .setHiveProperties(ImmutableMap.of("hive.dynamic-filtering-probe-blocking-timeout", "10s"))
```

Comment:
increase timeout for 1h to reduce risk of test flakyness

Reply:
done